lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2151,6 +2153,20 @@ static void transMetadataDecorations(Metadata *MD, SPIRVEntry *Target) {
           Target, Name->getString().str(), TypeKind));
       break;
     }
+    case spv::internal::DecorationHostAccessINTEL: {
+      assert(NumOperands == 3 &&
+             "HostAccessINTEL requires exactly 3 extra operand");
```

Comment:
Yeah, here's a bit incorrect naming/numbering of operands - it actually means that we expect metadata to contain 3 operands like `!{i32 6147, i32 0, !{"Name"}}`, where the decoration itself contains exactly 2 operands.
It was done alike to DecorationLinkageAttributes handling (see above).
I suppose the warning in the assert should be changed to avoid confusion.